disnake/partial_emoji.py

```diff
@@ -271,3 +272,40 @@ async def read(self) -> bytes:
             raise TypeError("PartialEmoji is not a custom emoji")
 
         return await super().read()
+
+    @staticmethod
+    def _to_name_id(
+        emoji: Optional[Union[str, Emoji, PartialEmoji]]
+    ) -> Tuple[Optional[str], Optional[int]]:
+        # API only supports exactly one of `name` and `id` being set
+        if isinstance(emoji, _EmojiTag):
+            if emoji.id:
+                return None, emoji.id
+            else:
+                return emoji.name, None
+        else:
+            return emoji, None
+
+    @staticmethod
+    def _from_name_id(
+        name: Optional[str], id: Optional[int], *, state: ConnectionState
+    ) -> Optional[Union[Emoji, PartialEmoji]]:
+        # NOTE: id may also be `0`
+
+        if not (name or id):
+            return None
+
+        emoji: Optional[Union[Emoji, PartialEmoji]] = None
+        if id:
+            emoji = state.get_emoji(id)
+        if not emoji:
+            emoji = PartialEmoji.with_state(
+                state=state,
+                # TODO: this renders correctly, but apparently shouldn't be done
+                # according to https://github.com/discord/discord-api-docs/pull/4983
+                name=name or "_",
```

Comment:
while it shouldn't be done, we should at least follow the emoji name rule of 2-32 characters.
```suggestion
                name=name or "__",
```